components/omega/test/ocn/EosTest.cpp

```diff
@@ -0,0 +1,494 @@
+   Array2DReal Tarray1  = Array2DReal("Tarray1", 1, 1);
+   Array2DReal Sarray2  = Array2DReal("Sarray2", 1, 1);
+   Array2DReal Tarray2  = Array2DReal("Tarray2", 1, 1);
+   const int K          = 0;
+   const int ICell      = 0;
+   Array2DReal SpecVol1 = Array2DReal("SpecVol1", 1, 1);
+   Array2DReal SpecVol2 = Array2DReal("SpecVol2", 1, 1);
+   Array2DReal SpecVol3 = Array2DReal("SpecVol3", 1, 1);
+   Array2DReal SpecVol4 = Array2DReal("SpecVol4", 1, 1);
+
+   Sarray1(0, 0)  = Sa;
+   Tarray1(0, 0)  = 15.;
+   Sarray2(0, 0)  = 33.;
+   Tarray2(0, 0)  = Ct;
+   SpecVol1(0, 0) = 0.0;
+   SpecVol2(0, 0) = 0.0;
+   SpecVol3(0, 0) = 0.0;
+   SpecVol4(0, 0) = 0.0;
+
+   LinearEOS specvollinear;
+   specvollinear(SpecVol1, ICell, K, Tarray1, Sarray1);
+   specvollinear(SpecVol2, ICell, K, Tarray2, Sarray1);
+   specvollinear(SpecVol3, ICell, K, Tarray1, Sarray2);
+   specvollinear(SpecVol4, ICell, K, Tarray2, Sarray2);
+   Real DRhoT  = 1. / SpecVol2(0, 0) - 1. / SpecVol1(0, 0);
+   Real DRhoS  = 1. / SpecVol3(0, 0) - 1. / SpecVol1(0, 0);
+   Real DRhoTS = 1. / SpecVol4(0, 0) - 1. / SpecVol1(0, 0);
+   // LOG_INFO("linearDensityLinearityTest: produced SpecVol from linear EOS");
+   bool Check1 = isApprox(DRhoS, ExpDS, RTol);
+   if (!Check1) {
+      Err++;
+      LOG_ERROR("linearDensityLinearityTest: DRhoS {}; expected {}", DRhoS,
+                ExpDS);
+   }
+   bool Check2 = isApprox(DRhoT, ExpDT, RTol);
+   if (!Check2) {
+      Err++;
+      LOG_ERROR("linearDensityLinearityTest: DRhoT {}; expected {}", DRhoT,
+                ExpDT);
+   }
+   bool Check3 = isApprox(DRhoTS, DRhoS + DRhoT, RTol);
+   if (!Check3) {
+      Err++;
+      LOG_ERROR("linearDensityLinearityTest: Sum(DRho) {}, DRhoTS {}",
+                DRhoS + DRhoT, DRhoTS);
+   }
+   // LOG_INFO("linearDensityLinearityTest: Sum(DRho) is undistinguishable from"
+   //          "DRhoTS");
+   if (Err == 0) {
+      LOG_INFO("linearDensityLinearityTest: PASS");
+   }
+   return Err;
+}
+
+// the main test (all in one to have the same log)
+// Single value tests:
+// --> one test calls the external GSW-C library
+// and compares the specific volume to the published value
+// --> next tests call the local Poly75t TEOS-10 calc
+// --> next tests call the linear Eos
+// Full array tests:
+// --> one tests the initialization/retrieval of Eos
+// --> next checks the value on a Eos with linear option
+// --> next checks the value on a Eos with Teos-10 option
+int eosTest(const std::string &MeshFile = "OmegaMesh.nc") {
+   int Err = initEosTest(MeshFile);
+   if (Err != 0) {
+      LOG_CRITICAL("EosTest: Error initializing");
+   }
+   const auto &Mesh = HorzMesh::getDefault();
+   LOG_INFO("Single value checks:");
+   Err += gswcSpecVolCheckValue();
+   Err += fetchCoeff();
+   // Err += poly75tDeltaCheckValue();
+   // Err += poly75tSpecVolCheckValue();
+   // Err += linearSpecVolCheckValue();
+   // Err += linearDensityLinearityTest();
```

Comment:
I think we can eliminate these since they are tested in the full array checks.

Reply:
Done!